pkgs/development/compilers/avian/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, zlib, jdk }:
+
+stdenv.mkDerivation {
+  name = "avian-1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "readytalk";
+    repo = "avian";
+    rev = "72c0547f4bdedc1ba8da2882317e3f758deab23f";
```

Comment:
From a conceptual pow, the important part is the content address (the `sha256` in this case); the url is only a hint of where to find the object with that content address. I like to imagine that there's an infinite, immutable store of objects that we're pulling from. Doesn't work out that way in reality, of course, but still.